File: android/ValleCOM/app/src/main/java/com/valleapp/vallecom/adaptadores/AdaptadorBuscarPedidos.java
package com.valleapp.vallecom.adaptadores;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageButton;
import android.widget.TextView;

import com.valleapp.vallecom.R;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

/**
 * Created by valle on 14/09/14.
 */
public class AdaptadorBuscarPedidos extends ArrayAdapter<JSONObject> {

    Context cx;
    List<JSONObject> values;

    public AdaptadorBuscarPedidos(Context context, List<JSONObject> obj) {
        super(context, R.layout.linea_pedido_externo, obj);
        this.cx = context;this.values= obj;
      }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

         LayoutInflater inflater = (LayoutInflater) cx
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View rowView = inflater.inflate(R.layout.linea_pedido_externo, parent, false);

        try {
            TextView nombre = rowView.findViewById(R.id.lblNombre);
            TextView cantidad = rowView.findViewById(R.id.lblCantidad);
            JSONObject art = values.get(position);
            nombre.setText(String.format("%s - %s",art.getString("Descripcion"), art.getString(("nomMesa"))));
            cantidad.setText(String.format("%s",art.getString("Can")));

            ImageButton btnRm = rowView.findViewById(R.id.btnBorrarPedido);
            if (!art.getString("servido").equals("0")) btnRm.setVisibility(View.GONE);
            btnRm.setTag(art);


        } catch (JSONException e) {
            e.printStackTrace();
        }


        return rowView;
    }
}

File: android/ValleTPV/ValleCOM/src/main/java/com/valleapp/vallecom/adaptadores/AdaptadorBuscarPedidos.kt
package com.valleapp.vallecom.adaptadores

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.valleapp.valletpv.R
import org.json.JSONException
import org.json.JSONArray
import org.json.JSONObject

class AdaptadorBuscarPedidos(
    private val context: Context,
    private val values: JSONArray
) : RecyclerView.Adapter<AdaptadorBuscarPedidos.PedidosViewHolder>() {

    inner class PedidosViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val nombreTextView: TextView = view.findViewById(R.id.lblNombre)
        val cantidadTextView: TextView = view.findViewById(R.id.lblCantidad)
        val borrarButton: ImageButton = view.findViewById(R.id.btnBorrarPedido)
        val lineaArtLayout: LinearLayout = view.findViewById(R.id.btnReenviarLinea)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PedidosViewHolder {
        val view =
            LayoutInflater.from(context).inflate(R.layout.linea_pedido_externo, parent, false)
        return PedidosViewHolder(view)
    }

    override fun onBindViewHolder(holder: PedidosViewHolder, position: Int) {
        try {
            val item: JSONObject = values.getJSONObject(position)
            holder.nombreTextView.text = buildString {
                append(item.getString("Descripcion"))
                append(" - ")
                append(item.getString("nomMesa"))
            }
            holder.cantidadTextView.text = item.getString("Can")

            if (item.getString("servido") != "0") {
                holder.borrarButton.visibility = View.GONE
            } else {
                holder.borrarButton.visibility = View.VISIBLE
                holder.borrarButton.tag = item
            }

            // Guarda el item en el tag de lineaArtLayout
            holder.lineaArtLayout.tag = item
        } catch (e: JSONException) {
            Log.e("AdaptadorBuscarPedidos", "Error al enlazar datos: ${e.message}", e)
        }
    }

    override fun getItemCount(): Int {
        return values.length()
    }


}
